roles/build_openstack_packages/tasks/install_dlrn.yml

```diff
@@ -93,29 +93,30 @@
       retries: 3
       delay: 5
 
-    - name: Install DLRN
-      ansible.builtin.shell: >
-        source {{ cifmw_bop_build_repo_dir }}/dlrn-venv/bin/activate;
-        pip install -r requirements.txt;
-        python3 setup.py install;
-      args:
-        chdir: '{{ cifmw_bop_build_repo_dir }}/DLRN'
+    - name: Install DLRN requirements
+      ansible.builtin.pip:
+        virtualenv: "{{ cifmw_bop_dlrn_venv }}"
+        requirements: "{{ cifmw_bop_build_repo_dir }}/DLRN/requirements.txt"
+
+    - name: Install DLRN from source
+      ansible.builtin.command:
+        cmd: "{{ cifmw_bop_dlrn_venv }}/bin/python setup.py install"
+        chdir: "{{ cifmw_bop_build_repo_dir }}/DLRN"
+        creates: "{{ cifmw_bop_dlrn_venv }}/bin/dlrn"
 
 - name: Install DLRN from pip
   when: not cifmw_bop_dlrn_from_source|bool
   block:
     - name: Pip install rdopkg
```

Comment:
Why not use a single pip task? It accepts a list of packages. Should be a bit faster.

Reply:
Yes, correct, Done!